### PR TITLE
fix(curriculum): cat-photo-app check type attribute

### LINKED ISSUE #47106: Cat Photo App 48 - Tests are not checking for radio buttons type attribute
### Describe the Issue

The tests aren't checking for the `type` attribute for both `radio` buttons, on [step 48](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-48) of the Cat Photo App. 
I can successfully pass this step with no `type` attributes set.

image
Code to replicate the issue :
```
      <label><input id="indoor" value="indoor" name="indoor-outdoor"> Indoor</label>
      <label><input id="outdoor" value="outdoor" name="indoor-outdoor"> Outdoor</label>
```

**Other affected pages :**
- [Step 45](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-45)
- [Step 47](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-47)

## [Step 45](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-45) :

image

Code to replicate the issue :
```
       <label><input id="indoor"> Indoor</label>
```

## [Step 47](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-47) :

image

Code to replicate the issue :
```
          <label><input id="indoor" name="indoor-outdoor"> Indoor</label>
          <label><input id="outdoor" name="indoor-outdoor"> Outdoor</label>
```

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-48

### Your code

```
 <label><input id="indoor" value="indoor" name="indoor-outdoor"> Indoor</label>
 <label><input id="outdoor" value="outdoor" name="indoor-outdoor"> Outdoor</label>
```

### Expected behavior

It would be better if the tests check for the `type` attribute, just like in the following steps :
- [Step 43](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-43)
- [Step 44](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-44)

## [Step 43 ](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-43):
image

## [Step 44](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-44)

image

### Screenshots

_No response_

### System

- Device: Laptop
- OS: Windows 10
- Browser: Chrome
- Version: 103.0.5060.134

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kelvinyy**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #47106

### CODE REVIEW

**Review by Sboonny — CHANGES_REQUESTED** [review]
Thank you for adding the test and taking care of it, but the campers who fail the test won't know that they are required to put in `type` attribute

Can you change this into its own hints, something like this

> Both radio buttons should have a `type` attribute. Check that there is a space after the opening tag's name and/or there are spaces before all attribute names.
>
> ```js
> const radioButtons = [...document.querySelectorAll('input[type="radio"]')];
> assert(radioButtons.every((btn) => btn.hasAttribute('type')));
> ```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5ef9b03c81a63668521804de.md [hunk 1]
```diff
@@ -30,6 +30,15 @@ Both radio buttons should have a `name` attribute. Check that there is a space a
 ```js
 const radioButtons = [...document.querySelectorAll('input[type="radio"]')];
 assert(radioButtons.every((btn) => btn.hasAttribute('name')));
+
```
Comment (Sboonny) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5ef9b03c81a63668521804de.md:33]:
```suggestion
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5ef9b03c81a63668521804de.md [hunk 2]
```diff
@@ -30,6 +30,15 @@ Both radio buttons should have a `name` attribute. Check that there is a space a
...
+```
+
+Both radio buttons should have a `type` attribute. Check that there is a space after the opening tag's name and/or there are spaces before all attribute names.
+
+```js
+const radioButtons = [...document.querySelectorAll('input')];
+assert(radioButtons.every((btn) => btn.hasAttribute('type')));
+
```
Comment (Sboonny) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5ef9b03c81a63668521804de.md:41]:
```suggestion
```

Comment (Sboonny) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5ef9b03c81a63668521804de.md:41]:
```suggestion
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5f1a80975fc4bcae0edb3497.md [hunk 3]
```diff
@@ -30,6 +30,16 @@ Both radio buttons should have a `value` attribute. Check that there is a space
 ```js
 const radioButtons = [...document.querySelectorAll('input[type="radio"]')];
 assert(radioButtons.every((btn) => btn.hasAttribute('value')));
+
+```
+Both radio buttons should have a `type` attribute. Check that there is a space after the opening tag's name and/or there are spaces before all attribute names.
```
Comment (Sboonny) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5f1a80975fc4bcae0edb3497.md:35]:
```suggestion

Both radio buttons should have a `type` attribute. Check that there is a space after the opening tag's name and/or there are spaces before all attribute names.
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5f1a80975fc4bcae0edb3497.md [hunk 4]
```diff
@@ -30,6 +30,16 @@ Both radio buttons should have a `value` attribute. Check that there is a space
 ```js
 const radioButtons = [...document.querySelectorAll('input[type="radio"]')];
 assert(radioButtons.every((btn) => btn.hasAttribute('value')));
+
```
Comment (Sboonny) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5f1a80975fc4bcae0edb3497.md:33]:
```suggestion
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5f1a80975fc4bcae0edb3497.md [hunk 5]
```diff
@@ -30,6 +30,16 @@ Both radio buttons should have a `value` attribute. Check that there is a space
...
+```
+Both radio buttons should have a `type` attribute. Check that there is a space after the opening tag's name and/or there are spaces before all attribute names.
+
+```js
+const radioButtons = [...document.querySelectorAll('input')];
+assert(radioButtons.every((btn) => btn.hasAttribute('type')));
+
+
+
```
Comment (Sboonny) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5f1a80975fc4bcae0edb3497.md:42]:
```suggestion
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5f1a80975fc4bcae0edb3497.md [hunk 6]
```diff
@@ -30,6 +30,16 @@ Both radio buttons should have a `value` attribute. Check that there is a space
...
 assert(radioButtons.every((btn) => btn.hasAttribute('value')));
+
+```
+Both radio buttons should have a `type` attribute. Check that there is a space after the opening tag's name and/or there are spaces before all attribute names.
+
+```js
+const radioButtons = [...document.querySelectorAll('input')];
+assert(radioButtons.every((btn) => btn.hasAttribute('type')));
+
+
```
Comment (Sboonny) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5f1a80975fc4bcae0edb3497.md:41]:
```suggestion
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5f1a80975fc4bcae0edb3497.md [hunk 7]
```diff
@@ -30,6 +30,16 @@ Both radio buttons should have a `value` attribute. Check that there is a space
...
 const radioButtons = [...document.querySelectorAll('input[type="radio"]')];
 assert(radioButtons.every((btn) => btn.hasAttribute('value')));
+
+```
+Both radio buttons should have a `type` attribute. Check that there is a space after the opening tag's name and/or there are spaces before all attribute names.
+
+```js
+const radioButtons = [...document.querySelectorAll('input')];
+assert(radioButtons.every((btn) => btn.hasAttribute('type')));
+
```
Comment (Sboonny) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5f1a80975fc4bcae0edb3497.md:40]:
```suggestion
```

**Review by Sboonny — CHANGES_REQUESTED** [review]
Wow, thank you for the quick update.

Nitpicks about the formating

**Comment by Sboonny** [comment]
I have fixed two spaced I didn't notice first, after the test have passed I will approve it. So

#### :sparkles: :tada: **AWESOME!** :tada: :sparkles:

 Hi @kelvinyy, thanks for this pull request and for contributing to the code-base for the first time. We are looking forward to more contributions from you in the future.

 Cheers & happy contributing!

**Review by Sembauke — APPROVED** [review]

**Review by Sboonny — APPROVED** [review]

**Pull request merged**